### PR TITLE
Add if and else_if

### PR DESCRIPTION
**Pull request by wxwisiasdf**

### CODE REVIEW

**Comment by schombert** [comment]
since you are adding codepoints, you need to bump the serialization version

File: src/scripting/effects.cpp [hunk 1]
```diff
@@ -4,8 +4,14 @@
...
 
+struct effect_local_state {
+	static constexpr inline uint32_t max_if_depth = 16;
+	bool if_is_triggered[max_if_depth] = { false };
+	uint32_t if_depth = 0;
+};
+
 #define EFFECT_PARAMTERS                                                                                                         \
-	uint16_t const *tval, sys::state &ws, int32_t primary_slot, int32_t this_slot, int32_t from_slot, uint32_t r_hi, uint32_t r_lo
+	uint16_t const *tval, sys::state &ws, int32_t primary_slot, int32_t this_slot, int32_t from_slot, uint32_t r_hi, uint32_t r_lo, effect_local_state els
```
Comment (schombert) [inline src/scripting/effects.cpp:14]: Shouldn't this be passed by reference?

Comment (wxwisiasdf) [reply src/scripting/effects.cpp:14]: oversight, whoops

File: src/scripting/effects.cpp [hunk 2]
```diff
@@ -15,14 +15,44 @@ inline uint32_t apply_subeffects(EFFECT_PARAMTERS) {
...
+}
+uint32_t es_else_if_scope(EFFECT_PARAMTERS) {
+	uint32_t ret = 0;
+	if((tval[0] & effect::scope_has_limit) != 0) {
+		auto limit = trigger::payload(tval[2]).tr_id;
+		if(!els && trigger::evaluate(ws, limit, primary_slot, this_slot, from_slot)) {
+			ret = apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo, els);
+			els = true;
+		}
+	} else {
```
Comment (schombert) [inline src/scripting/effects.cpp:51]: needs to test for !els here, otherwise else will always run if there is no limit in it

**Pull request merged**